crates/components-runtime/src/config.rs

```diff
@@ -0,0 +1,80 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct ComponentsConfiguration {
+    #[serde(default = "default_data_collection_components")]
+    pub data_collection: Vec<DataCollectionComponents>,
+    #[serde(default = "default_consent_mapping_components")]
+    pub consent_mapping: Vec<ConsentMappingComponents>,
+    pub cache: Option<PathBuf>,
+}
+
+fn default_data_collection_components() -> Vec<DataCollectionComponents> {
+    vec![]
+}
+
+fn default_consent_mapping_components() -> Vec<ConsentMappingComponents> {
+    vec![]
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct DataCollectionComponents {
+    pub name: String,
+    pub component: String,
+    pub credentials: HashMap<String, String>,
+    #[serde(default = "default_data_collection_component_config")]
+    pub config: DataCollectionComponentConfig,
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+pub struct DataCollectionComponentConfig {
+    #[serde(default = "default_true")]
+    pub anonymization: bool,
+    #[serde(default = "default_pending")]
+    pub default_consent: String,
+    #[serde(default = "default_true")]
+    pub track_event_enabled: bool,
+    #[serde(default = "default_true")]
+    pub user_event_enabled: bool,
+    #[serde(default = "default_true")]
+    pub page_event_enabled: bool,
+}
```

Comment:
I think you can replace the field-specific serde default attributes by a struct-global serde default attribute instead
It would use the `Default` trait of the struct (https://serde.rs/container-attrs.html#default )

```suggestion
#[serde(default)]
pub struct DataCollectionComponentConfig {
    pub anonymization: bool,
    pub default_consent: String,
    pub track_event_enabled: bool,
    pub user_event_enabled: bool,
    pub page_event_enabled: bool,
}
```